hail/python/hail/table.py

```diff
@@ -209,11 +209,11 @@ def aggregate(self, **named_exprs):
 
         base, cleanup = self._parent._process_joins(*group_exprs.values(), *named_exprs.values())
 
-        return Table._from_java(base._jt.keyByAndAggregate(
-            str(hl.struct(**named_exprs)._ir),
-            str(hl.struct(**group_exprs)._ir),
-            joption(self._npartitions),
-            self._buffer_size))
+        return cleanup(Table(TableKeyByAndAggregate(base._tir,
```

Comment:
`cleanup` wasn't being called before. Pretty sure it's not needed here (any temp fields added in the joins are gone after the keyByAndAggregate).